### PR TITLE
Added PrimitiveSettings + 3 overloads to Primitive::Create method

### PR DESCRIPTION
**Pull request by SmusiJarviss**

### CODE REVIEW

**Review by VALERA771 — COMMENTED** [review]
I don't really think that we need to put that method to obsolete and create another struct just for containing parameters. You don't use this struct anywhere except Create method

**Comment by NaoUnderscore** [comment]
> I don't really think that we need to put that method to obsolete and create another struct just for containing parameters. You don't use this struct anywhere except Create method

I don't really get what's wrong using a struct to better organize parameters and objects initialization.
It's pretty much safe, consistent and readable instead of manually specifying which parameters to pass, especially in case you need to spawn primitives in multiple lines and you want to save a "configuration" which can be your `PrimitiveSettings` struct.
Method has been marked obsolete because `Color?` as parameter changes the signature, hence you're forced to mark it obsolete.

File: Exiled.API/Features/Toys/Primitive.cs [hunk 1]
```diff
@@ -75,15 +75,68 @@ public bool Collidable
...
 
+        /// <summary>
+        /// Creates a new <see cref="Primitive"/>.
+        /// </summary>
+        /// <param name="position">The position of the <see cref="Primitive"/>.</param>
+        /// <param name="rotation">The rotation of the <see cref="Primitive"/>.</param>
+        /// <param name="scale">The scale of the <see cref="Primitive"/>.</param>
+        /// <param name="spawn">Whether or not the <see cref="Primitive"/> should be initially spawned.</param>
+        /// <returns>The new <see cref="Primitive"/>.</returns>
+        [Obsolete("Deprecated due to new methods. Create(Vector3, Vector3, Vector3, bool, Color)", true)]
```
Comment (NaoUnderscore) [inline Exiled.API/Features/Toys/Primitive.cs:86]: `Use Create(Vector3, Vector3, Vector3, bool, Color) instead.`

File: Exiled.API/Features/Toys/Primitive.cs [hunk 2]
```diff
@@ -75,15 +75,68 @@ public bool Collidable
...
+        /// <summary>
+        /// Creates a new <see cref="Primitive"/>.
+        /// </summary>
+        /// <param name="primitiveType">The type of primitive to spawn.</param>
+        /// <param name="position">The position of the <see cref="Primitive"/>.</param>
+        /// <param name="rotation">The rotation of the <see cref="Primitive"/>.</param>
+        /// <param name="scale">The scale of the <see cref="Primitive"/>.</param>
+        /// <param name="spawn">Whether or not the <see cref="Primitive"/> should be initially spawned.</param>
+        /// <returns>The new <see cref="Primitive"/>.</returns>
+        [Obsolete("Deprecated due to new methods. Create(PrimitiveType, Vector3, Vector3, Vector3, bool, Color)", true)]
```
Comment (NaoUnderscore) [inline Exiled.API/Features/Toys/Primitive.cs:112]: `Use Create(Vector3, Vector3, Vector3, bool, Color) instead.`

**Review by NaoUnderscore — CHANGES_REQUESTED** [review]
Add an overload to `Light::Create` as well.

File: Exiled.API/Features/Toys/Light.cs [hunk 3]
```diff
@@ -80,9 +81,10 @@ public bool ShadowEmission
         /// <param name="scale">The scale of the <see cref="Light"/>.</param>
         /// <param name="spawn">Whether the <see cref="Light"/> should be initially spawned.</param>
         /// <returns>The new <see cref="Light"/>.</returns>
+        [Obsolete("Use Create(Vector3, Vector3, Vector3, bool, Color) instead.", true)]
```
Comment (louis1706) [inline Exiled.API/Features/Toys/Light.cs:84]: You don't need to obsolete this

File: Exiled.API/Features/Toys/Primitive.cs [hunk 4]
```diff
@@ -75,15 +75,68 @@ public bool Collidable
...
+        /// <summary>
+        /// Creates a new <see cref="Primitive"/>.
+        /// </summary>
+        /// <param name="primitiveType">The type of primitive to spawn.</param>
+        /// <param name="position">The position of the <see cref="Primitive"/>.</param>
+        /// <param name="rotation">The rotation of the <see cref="Primitive"/>.</param>
+        /// <param name="scale">The scale of the <see cref="Primitive"/>.</param>
+        /// <param name="spawn">Whether or not the <see cref="Primitive"/> should be initially spawned.</param>
+        /// <returns>The new <see cref="Primitive"/>.</returns>
+        [Obsolete("Use Create(PrimitiveType, Vector3, Vector3, Vector3, bool, Color) instead.", true)]
```
Comment (louis1706) [inline Exiled.API/Features/Toys/Primitive.cs:112]: Same than above

File: Exiled.API/Features/Toys/Primitive.cs [hunk 5]
```diff
@@ -75,15 +75,68 @@ public bool Collidable
...
 
+        /// <summary>
+        /// Creates a new <see cref="Primitive"/>.
+        /// </summary>
+        /// <param name="position">The position of the <see cref="Primitive"/>.</param>
+        /// <param name="rotation">The rotation of the <see cref="Primitive"/>.</param>
+        /// <param name="scale">The scale of the <see cref="Primitive"/>.</param>
+        /// <param name="spawn">Whether or not the <see cref="Primitive"/> should be initially spawned.</param>
+        /// <returns>The new <see cref="Primitive"/>.</returns>
+        [Obsolete("Use Create(Vector3, Vector3, Vector3, bool, Color) instead.", true)]
```
Comment (louis1706) [inline Exiled.API/Features/Toys/Primitive.cs:86]: Same than above

**Review by louis1706 — CHANGES_REQUESTED** [review]
I am not 100% sure but default color is not white for nw but a specific color of gray

File: Exiled.API/Features/Toys/Light.cs (hunk 3 above)
Comment (NaoUnderscore) [reply Exiled.API/Features/Toys/Light.cs:84]: Signature won't match, compiler will ask you which method to use. Test it by yourself.

**Review by Misaka-ZeroTwo — APPROVED** [review]

File: Exiled.API/Features/Toys/Light.cs [hunk 6]
```diff
@@ -94,6 +96,31 @@ public static Light Create(Vector3? position = null, Vector3? rotation = null, V
...
+            Light light = new(UnityEngine.Object.Instantiate(ToysHelper.LightBaseObject));
+
+            light.AdminToyBase.transform.position = position ?? Vector3.zero;
+            light.AdminToyBase.transform.eulerAngles = rotation ?? Vector3.zero;
+            light.AdminToyBase.transform.localScale = scale ?? Vector3.one;
+
+            if (spawn)
+                light.Spawn();
+
+            light.Color = color ?? Color.white;
```
Comment (Thundermaker300) [inline Exiled.API/Features/Toys/Light.cs:119]:
```suggestion
            light.Color = color ?? Color.gray;
```
Should match NW's default

File: Exiled.API/Features/Toys/Primitive.cs [hunk 7]
```diff
@@ -95,6 +148,7 @@ public static Primitive Create(Vector3? position = null, Vector3? rotation = nul
                 primitive.Spawn();
 
             primitive.AdminToyBase.NetworkScale = primitive.AdminToyBase.transform.localScale;
+            primitive.Color = color ?? Color.white;
```
Comment (Thundermaker300) [inline Exiled.API/Features/Toys/Primitive.cs:151]:
```suggestion
            primitive.Color = color ?? Color.gray;
```
Should match NW's default

File: Exiled.API/Features/Toys/Primitive.cs [hunk 8]
```diff
@@ -121,6 +176,30 @@ public static Primitive Create(PrimitiveType primitiveType = PrimitiveType.Spher
 
             primitive.AdminToyBase.NetworkScale = primitive.AdminToyBase.transform.localScale;
             primitive.Base.NetworkPrimitiveType = primitiveType;
+            primitive.Color = color ?? Color.white;
```
Comment (Thundermaker300) [inline Exiled.API/Features/Toys/Primitive.cs:179]:
```suggestion
            primitive.Color = color ?? Color.gray;
```
Should match NW's default

**Review by Thundermaker300 — CHANGES_REQUESTED** [review]

File: Exiled.API/Features/Toys/Light.cs (hunk 3 above)
Comment (louis1706) [reply Exiled.API/Features/Toys/Light.cs:84]: we just don't need to remove this method ?

**Review by NaoUnderscore — APPROVED** [review]

**Pull request merged**